controllers/node_controller.go

```diff
@@ -111,9 +110,11 @@ func (r *nodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (resul
 			return ctrl.Result{}, fmt.Errorf("failed to create new nodeconfig: %w", err)
 		}
 
+		r.log.Info("rebooting instance")
```

Comment:
This shows up as:
```
INFO	controllers.node	rebooting instance
```
I suggest moving the log into `RebootAndReinitialize()` so that more context is added plus we know for sure that the reboot will be attempted.